Small spot_detection update (#20)
DragaDoncila/napari-scipy2025-workshop: napari-workshops/notebooks/spot_detection_functions.md

```

Notice that because we told `magicgui` that our function will use not just any numpy array, but 

> specifically `ImageData`---the data of an Image layer---and that it will also return that same type, `magicgui`

Reviewer: ```suggestion
specifically `ImageData`---the data of a `napari` Image layer---and that it will also return that same type, `magicgui`
```

gaussian_high_pass(viewer.layers['spots'].data)
```

```{important}

> Because we are just returning `ImageData`, so there is no information passed about colormaps, blending, etc. If we want to specify that, we would need to annotate as [`LayerDataTuple`](https://napari.org/stable/guides/magicgui.html#returning-napari-types-layerdatatuple). (We will do this in the next example.)

Reviewer: ```suggestion
Because we are just returning `ImageData`, there is no information passed about colormaps, blending, etc. If we want to specify that, we would need to annotate as [`LayerDataTuple`](https://napari.org/stable/guides/magicgui.html#returning-napari-types-layerdatatuple). (We will do this in the next example.)
```

```{tip}
Hover over the labels `image` and `sigma` -- the names of the parameters we passed to the function.
You should see tooltips with the docstring information! How cool is that? More *magic* from `magicgui`!
```

> Our `gaussian_high_pass` object *is the widget*, so we can easily get the value of the current setting:

Reviewer: ```suggestion
Our `magicgui` decorated `gaussian_high_pass` object *is the widget*, so we can easily get the value of the current setting:
```

## A more complex example

Finally, lets make a widget for the whole workflow developed in the [Exploratory analysis: spot detection](spot_detection_basic) 
notebook as a function. We will need to write a function and then properly annotate it such that `magicgui` can generate the widgets. 
This time we are also starting with image layer (data), but then we want a Points layer with points. We could again return 

> just the layer data using `napari.types.PointsData`. But lets get a nicer Points layer instead, so 

Reviewer: ```suggestion
just the layer data using `napari.types.PointsData`. However, we would like some control over the Points layer visualization, so 
```
